Composer/packages/client/src/shell/useShell.ts

```diff
@@ -20,7 +20,7 @@ const FORM_EDITOR = 'PropertyEditor';
 
 type EventSource = 'VisualEditor' | 'PropertyEditor';
 
-export function useShell(source: EventSource): { api: ShellApi; data: ShellData } {
+export function useShell(source?: EventSource): { api: ShellApi; data: ShellData } {
```

Comment:
See my comment above. The `useShell` hook is only intended to bind client state / actions to extension contexts. The `source` param should be required.